### PR TITLE
fix serveral issues and add more corrupted images

### PR DESCRIPTION
**Pull request by jaysim**

### CODE REVIEW

File: src/ntfsck.c [hunk 1]
```diff
@@ -878,6 +878,56 @@ static int ntfsck_add_inode_to_lostfound(ntfs_inode *ni, FILE_NAME_ATTR *fn,
...
+{
+	/* Do not delete system file */
+	if (mft_no < FILE_first_user)
+		return;
+
+	ntfs_bitmap_clear_bit(vol->mftbmp_na, mft_no);
+	ntfsck_mft_bmp_bit_clear(mft_no);
+}
+
+static void __ntfsck_delete_orphaned_inode(ntfs_inode **ni)
```
Comment (namjaejeon) [inline src/ntfsck.c]: any reason why __ is added as prefix ?

File: src/ntfsck.c [hunk 2]
```diff
@@ -878,6 +878,56 @@ static int ntfsck_add_inode_to_lostfound(ntfs_inode *ni, FILE_NAME_ATTR *fn,
 	return ret;
 }
 
+/* delete orphaned mft, call this when inode open failed. */
+static void __ntfsck_delete_orphaned_mft(ntfs_volume *vol, u64 mft_no)
```
Comment (namjaejeon) [inline src/ntfsck.c]: __ prefix? why?

File: src/ntfsck.c [hunk 3]
```diff
@@ -989,6 +1036,12 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 			parent_ni = ntfs_inode_open(vol, MREF(parent_no));
 
 		if (parent_ni) {
+			if (__ntfsck_cmp_parent_mft_sequence(parent_ni, fn)) {
```
Comment (namjaejeon) [inline src/ntfsck.c]: __ prefix ? Why?

File: src/ntfsck.c [hunk 4]
```diff
@@ -989,6 +1036,12 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
...
+				/* do not add inode to parent */
+				ntfs_log_info("##### Different seqnence number\n");
```
Comment (namjaejeon) [inline src/ntfsck.c:1097]: Probably we can add two sequence numbers in this print.

File: src/ntfsck.c (hunk 2 above)
Comment (jaysim) [reply src/ntfsck.c]: I removed it !

**Comment by namjaejeon** [comment]
Applied. Thanks!

**Comment by namjaejeon** [comment]
@jaysim the below corrupted image is wrong. mftmirror lcn number is bigger than ntfs.img size. It seems that the dump is not perfectly finished and appears to be aborted. So ntfsck can not read mftmirr mft record because the partition image is truncated halfway through.

NEED_TO_FIX/mftmirror_file_load_failure/ntfs.img.tar.xz

**Pull request closed without merging**